### ISSUE TITLE
Issue re-creation of API when it has default application quota

### ISSUE THREAD

**Issue by thotanaresh** [Axway-API-Management-Plus/apimanager-swagger-promote#86]
First, Thanks for the amazing tool!
Environment: Axway-7.5.3
Issue:
1. Create an api using the tool with the following config
```
{
   "name": "Test API",
   "path": "/api/v1.0",
   "state": "published",
   "version": "1.0.0",
   "apiDefinition": "../api.json",
   "organization": "API Development",
   "image": "../images/logo.png",
   "backendBasepath": "https://localhost",
   "securityProfiles": [
      {
         "name": "_default",
         "isDefault": true,
         "devices": [
            {
               "name": "API Key",
               "type": "apiKey",
               "order": 0,
               "properties": {
                  "apiKeyFieldName": "apikey",
                  "takeFrom": "HEADER",
                  "removeCredentialsOnSuccess": "true"
               }
            }
         ]
      }
   ],
   "applicationQuota": {
      "restrictions": [
         {
            "method": "*",
            "type": "throttle",
            "config": {
               "period": "seconds",
               "messages": 25,
               "per": 60
            }
         }
      ]
   }
}
```
2. Login to API Manager and Create an application and add access to the above created api
3. Change the swagger definition (eg. change any description to test redeployment using tool)
4. Deploy again using the tool is giving below error:
```
"Running API-Manager Promote version 1.5.2-1 ..."
0      INFO                   App| ------------------------------------------------------------------------
0      INFO                   App| API-Manager Promote Version: 1.5.2-1
1      INFO                   App|
1      INFO                   App| To report issues or get help, please visit:
8      INFO                   App| https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote
9      INFO                   App| ------------------------------------------------------------------------
26     INFO  nvironmentProperties| Loaded environment properties from file: env.properties.
27     INFO  nvironmentProperties| Loaded environment properties from file: env.test.properties.
1203   INFO  IImportConfigAdapter| Reading API-Definition (Swagger/WSDL) from file: 'C:\repo\api.json' (absolute path)
1247   INFO     APIManagerAdapter| API-Manager version is: 7.5.3
1323   INFO     APIManagerAdapter| Found existing API on path: '/api/v1.0' (published) (ID: '3f6a33e7-0ff3-4c3c-9511-1fda69c96b56')
1769   INFO        APIChangeState| Changed property: serviceProfiles[Desired: '{_default=ServiceProfile [apiId=null, basePath=https://localhost]}' vs Actual: '{_default=ServiceProfile [apiId=null, basePath=https://there.is.no.host]}']
1813   INFO     APIManagerAdapter| Strategy: Going to update existing API: Test API (Version: 1.0.0)
1813   INFO     APIManagerAdapter| Recognized the following changes. Potentially Breaking: [serviceProfiles] plus Non-Breaking: []
1814   INFO     APIManagerAdapter| Strategy: Apply breaking changes: [serviceProfiles] & and Non-Breaking: [], for PUBLISHED API. Recreating it!
1817   INFO     APIManagerAdapter| Importing backend API (Swagger/WSDL Import)
1919   INFO     APIManagerAdapter| Create API-Proxy (Front-End API)
3110   INFO     APIManagerAdapter| Updating proxy for the following properties: caCerts securityProfiles serviceProfiles path version name
3204   INFO     APIManagerAdapter| Updating API-Image from: ../images/logo.png
3278   INFO     APIManagerAdapter| Updating API-Status from: 'unpublished' to 'published'
3370   INFO     APIManagerAdapter| Actual API state set to: published
3371   INFO     APIManagerAdapter| Updating Application-Default-Quota for API: Test API
3531   INFO     APIManagerAdapter| Application-Default quota successfully updated: Test API
3533   INFO     APIManagerAdapter| Creating API-Access for the following apps: '[[Test1 App 1]]'
3645   INFO     APIManagerAdapter| Creating API-Access for the following apps: '[]'
3646   INFO     APIManagerAdapter| Granting access to newly created API
3737   INFO     APIManagerAdapter| Taking over existing quota config for application: 'Test1 App 1' to newly created API.
3756   ERROR          RestAPICall| Response: 'HTTP/1.1 400 Bad Request'
3756   ERROR                  App| Can't update application quota.
3757   ERROR                  App| Can't update application quota.
com.axway.apim.lib.AppException: Can't update application quota.
        at com.axway.apim.actions.tasks.UpgradeAccessToNewerAPI.execute(UpgradeAccessToNewerAPI.java:90)
        at com.axway.apim.actions.RecreateToUpdateAPI.execute(RecreateToUpdateAPI.java:44)
        at com.axway.apim.swagger.APIManagerAdapter.applyChanges(APIManagerAdapter.java:162)
        at com.axway.apim.App.run(App.java:166)
        at com.axway.apim.App.main(App.java:43)
```

**Comment by cwiechmann** [comment]
@thotanaresh, I made some significant changes in the App-Handling area esp. for API-Re-Creation and created an IntegrationTest simulating what you have done. The test is now working fine.
Would you be okay to test a Pre-Release if it is now working on your side as expected? I would share the release in this issue with you, once all IntegrationTests are stable. Just let me know.

**Comment by thotanaresh** [comment]
> @thotanaresh, Thanks for reporting this use-case. One question: The app you have created in the UI, belongs it to the same organization as the API or is it in a foreign organization?

Yes, app belongs to the same organization.

**Comment by thotanaresh** [comment]
> @thotanaresh, I made some significant changes in the App-Handling area esp. for API-Re-Creation and created an IntegrationTest simulating what you have done. The test is now working fine.
> Would you be okay to test a Pre-Release if it is now working on your side as expected? I would share the release in this issue with you, once all IntegrationTests are stable. Just let me know.

Sure, I can test from my end for that use case. Thanks

**Comment by thotanaresh** [comment]
@cwiechmann I have created new package from your code and tried in my environment, still having the same issue. 
NOTE: I have changed the `period` in the configuration to `second` instead of `seconds`, Otherwise you get 500 response for API request.

**Comment by cwiechmann** [comment]
In that case, please tell me, is the app belonging to the same org as the API itself and have you configured app-quota override for this app&api?

**Comment by cwiechmann** [comment]
And please setup debug, re-run you scenario and share the log-file here. Thanks!

https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/wiki/7.2-Enable-Debug

**Comment by thotanaresh** [comment]
@cwiechmann Yes, the app and the API belongs to same organization.
* With app-quota override for that API - PASSED
* Without app-quota override (Still have default quota) - FAILED

**Comment by cwiechmann** [comment]
@thotanaresh, confirmed the error. I will extend the Test-Case and fix it.

**Comment by cwiechmann** [comment]
Now the issue is fixed. The reason was, that the API-Manager REST-API also returns Default-Quotas when asking for Application-Quota. This makes sense at runtime when validating if an app can still call an API. 

But Swagger-Promote tried to take over the returned default Quota as an Application quota, which failed. Now the code taking it over, ignores Application- or System-Default-Quotas.
I extended the Test-Case to validate expected behavior for the future.

**Comment by thotanaresh** [comment]
@cwiechmann Thanks mate! I have tested from my end, and is working perfect.

**Comment by cwiechmann** [comment]
Released with version 1.5.3
